src/test/java/org/javarosa/core/model/instance/geojson/GeojsonGeometryTest.java

```diff
@@ -0,0 +1,37 @@
+package org.javarosa.core.model.instance.geojson;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class GeojsonGeometryTest {
```

Comment:
We could also add a test (and implementation) for validating that Polygons always have identical first and last points. I'm not sure that that makes sense however: Collect will have to dynamically detect a polygon by checking for a match anyway, so invalid polygons would just end up being represented as traces.

Reply:
Does Collect check that first and last match before dropping last? If not, it probably should.

Reply:
There's no implementation for showing shapes (it's a separate issue from showing traces/lines), but yes it'll need to do that.

Reply:
There is an implementation that shows existing shapes (defaults or previously-selected). But anyway, sounds like that's on your radar and you'll consolidate all that anyway.